kong/plugins/hmac-auth/daos.lua

```diff
@@ -17,6 +17,7 @@ return {
       { consumer = { type = "foreign", reference = "consumers", required = true, on_delete = "cascade", }, },
       { username = { type = "string", required = true, unique = true }, },
       { secret = { type = "string", auto = true }, },
+      { public_key = { type = "string" }, },
```

Comment:
This is a breaking change for version compatibility with older data planes; updates will need to be made to the [compatibility layer ](https://github.com/Kong/kong/blob/master/kong/clustering/compat/removed_fields.lua)to handle these changes ahead of time of before a release.

Reply:
I'm not sure to understand what needs to be updated. You linked to a file named `removed_fields.lua` but no field has been removed, only added. I've checked in the folder if there was a file named `added_fields.lua` but could not find any. I've also checked other PR doing similar changes on the configuration and could not find any modification related to the compatibility layer.

Could you provide more informations on what needs to be done exactly ?

Reply:
@mideuger This is something that will need to be handled closer to release time when this feature is targeting a particular version of Kong Gateway; I just wanted to mark this PR as something that is going to need further testing and updates when it comes to release time.

> I've also checked other PR doing similar changes on the configuration and could not find any modification related to the compatibility layer.

Since you are adding a new field to the `hmac-auth` plugin, older data planes will not know how to process this field when it is transmitted in the payload in hybrid mode. To allow for newer control planes to interoperate with older data planes the field `public_key` needs to be removed before the payload is sent down.

Reply:
hi @mideuger

The comment @mikefero means that you need to add your new field "enforce_unique_consumer" into this file removed_fields.lua, here is an example of how to add a new field to plugin conf and make it compatitble with old version: https://github.com/Kong/kong/pull/11523